admin/deployments.go

```diff
@@ -353,38 +354,37 @@ func (s *Service) HibernateDeployments(ctx context.Context) error {
 }
 
 func (s *Service) teardownDeployment(ctx context.Context, depl *database.Deployment) error {
-	// Connect to the deployment's runtime
-	rt, err := s.openRuntimeClientForDeployment(depl)
+	// Delete the deployment
+	err := s.DB.DeleteDeployment(ctx, depl.ID)
 	if err != nil {
 		return err
 	}
-	defer rt.Close()
 
-	// Delete the instance
-	_, err = rt.DeleteInstance(ctx, &runtimev1.DeleteInstanceRequest{
-		InstanceId: depl.RuntimeInstanceID,
-	})
+	// Connect to the deployment's runtime and delete the instance
+	rt, err := s.openRuntimeClientForDeployment(depl)
 	if err != nil {
-		return err
+		s.Logger.Error("failed to open runtime client", zap.String("deployment_id", depl.ID), zap.String("runtime_instance_id", depl.RuntimeInstanceID), zap.Error(err), observability.ZapCtx(ctx))
+	} else {
+		_, err = rt.DeleteInstance(ctx, &runtimev1.DeleteInstanceRequest{
+			InstanceId: depl.RuntimeInstanceID,
+		})
+		if err != nil {
+			s.Logger.Error("failed to delete instance", zap.String("deployment_id", depl.ID), zap.String("runtime_instance_id", depl.RuntimeInstanceID), zap.Error(err), observability.ZapCtx(ctx))
+		}
 	}
+	defer rt.Close()
```

Comment:
This can cause a panic if `err != nil` – it will try to call `.Close()` on `rt` even if it is `nil`.

Maybe move it to the top of the `else` clause? (`defer` is scoped by function, not by inner scopes like if/else)

Reply:
Fixed - thanks